src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -159,6 +164,12 @@ private[health] class HealthCheckActor(
     }
   }
 
+  def checkEnoughInstancesRunning(): Boolean = {
+    val activeInstances = Await.result(instanceTracker.countActiveSpecInstances(app.id), 5 seconds)
```

Comment:
we should handle throwing case here

Reply:
Agree , when the await timeout, you should add a logging line related to this.
You could even add a --conservative option to enforce to not kill instances even if this timeout